discovery-provider/src/tasks/index_spl_token.py

```diff
@@ -146,14 +149,19 @@ def decode_memo_and_extract_vendor(memo_encoded: str) -> str:
 
 def parse_spl_token_transaction(
     solana_client_manager: SolanaClientManager,
-    tx_sig: ConfirmedSignatureForAddressResult,
+    tx_sig: str,
 ) -> Optional[SplTokenTransactionInfo]:
     try:
-        tx_info = solana_client_manager.get_sol_tx_info(tx_sig["signature"])
-        result = tx_info["result"]
-        meta = result["meta"]
-        error = meta["err"]
-        if error:
+        tx_info = solana_client_manager.get_sol_tx_info(tx_sig)
+        result = tx_info.value
+        if not result:
+            return None
+
+        meta = result.transaction.meta
+        if not meta:
+            return None
```

Comment:
Might wanna add logs for these cases, don't think we expect this to happen so would be good to know if it did.

Reply:
raising, catching & logging as error!